hack/generator/pkg/astmodel/go_file_writer.go

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astmodel
+
+import (
+	"bufio"
+	"bytes"
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astbuilder"
+	"go/ast"
+	"go/format"
+	"go/parser"
+	"go/token"
+	"io"
+	"k8s.io/klog/v2"
+	"strings"
+)
+
+// GoFileWriter encapsulates the work we do to write a Go AST into a specific file
+type GoFileWriter struct {
+	name         *ast.Ident
+	declarations []ast.Decl
+}
```

Comment:
It's possible this abstraction will completely go away with the changes @Porges is making in PR #336

Reply:
Yup. It did.